filecoindot/src/tests/ocw.rs

```diff
@@ -125,10 +125,10 @@ fn should_submit_vote_in_ocw() {
             assert_eq!(tx.signature, Some((i as u64, ())));
             assert_eq!(
                 tx.call,
-                Call::FileCoinModule(crate::Call::submit_block_vote(
-                    resp.cids[i].inner.to_vec(),
-                    resp.blocks[i].messages.inner.to_vec()
-                ))
+                Call::FileCoinModule(crate::Call::submit_block_vote {
```

Comment:
@willeslau this looks neat, Is this one of the features in rust2021? What's the proper name for this new way of calling a function? I may have missed to read the release notes in the new verions of rust.

Reply:
this is part of the syntax of substrate LOL

Reply:
This is interesting because rust doesn't provide named parameters yet, (and not sure it would be in the future) but somehow substrate is able to achieve that here.

```rust
        #[pallet::weight(T::WeightInfo::submit_block_vote())]
        pub fn submit_block_vote(
            origin: OriginFor<T>,
            block_cid: BlockCid,
            message_root_cid: MessageRootCid,
        ) -> DispatchResult {
```

Reply:
After expanding the code, the `Call` becomes an `enum` and with it `submit_block_vote` is a struct inside of this enum. 
```rust
    #[codec(encode_bound())]
    #[codec(decode_bound())]
    #[scale_info(skip_type_params(T), capture_docs = "always")]
    #[allow(non_camel_case_types)]
    pub enum Call<T: Config> {
        #[doc(hidden)]
        #[codec(skip)]
        __Ignore(
            frame_support::sp_std::marker::PhantomData<(T,)>,
            frame_support::Never,
        ),
        /// Adds a new relayer to the relayer set.
        add_relayer { v: T::AccountId },
        /// Removes an existing relayer from the set.
        remove_relayer { v: T::AccountId },
        /// Update the vote threshold
        set_vote_threshold { threshold: u32 },
        /// Commits a vote in favour of the provided block cid and message root.
        submit_block_vote {
            block_cid: BlockCid,
            message_root_cid: MessageRootCid,
        },
        /// Admin can close the proposal when it has expired. The admin ought to have called this
        /// when the proposal expired, otherwise it
        close_block_proposal { block_cid: BlockCid },
        /// Verify the receipt of the filecoin
        verify_receipt { proof: Vec<Vec<u8>>, cid: Vec<u8> },
        /// Verify the state of the filecoin
        verify_state { proof: Vec<Vec<u8>>, cid: Vec<u8> },
    }
```